tests/unit/specification/grammar_test.py

```diff
@@ -73,7 +74,7 @@ def parse_unproven_session(string: str) -> model.UnprovenSession:
     if diagnostics_to_error(unit.diagnostics, error, STDIN):
         error.propagate()
     assert isinstance(unit.root, lang.SessionDecl)
-    return create_unproven_session(unit.root, ID("Package"), Path("<stdin>"))
+    return create_unproven_session(unit.root, ID("Package"), Path("<stdin>"), error)
```

Comment:
`error.propagate()`  should be called after `create_*`, otherwise any error in `create_*` will be just ignored.

Reply:
Fixed